File: C4/prg/logistic_company/retarded/Car.java
package logistic_company.retarded;

public class Car {
    private int id;
    private double carryWeight;
    private double weight;

    public Car(int id, double carryWeight, double weight) {
        this.id = id;
        this.carryWeight = carryWeight;
        this.weight = weight;
    }

    public int getId() {
        return this.id;
    }

    public double getCarryWeight() {
        return this.carryWeight;
    }

    public double getWeight() {
        return this.weight;
    }

    public void setWeight(double weight) {
        this.weight = weight;
    }
}

File: C4/PRG/logistic_company/correct/Car.kt
package logistic_company.correct

open class Car(val _carryWeight: Double, var _weight: Double = 0.0) {

    val carryWeight: Double
        get() { return field }

    var weight: Double = 0.0
        get() { return field }

        set(value) {
            field += value
        }

    init {
        carryWeight = _carryWeight
        weight = _weight
    }
    
}